go-controller/pkg/ovn/namespace.go

```diff
@@ -409,21 +410,44 @@ func (oc *Controller) deleteNamespaceLocked(ns string) *namespaceInfo {
 }
 
 func (oc *Controller) createNamespaceAddrSetAllPods(ns string) (addressset.AddressSet, error) {
-	// Get all the pods in the namespace and append their IP to the address_set
 	var ips []net.IP
-	existingPods, err := oc.watchFactory.GetPods(ns)
-	if err != nil {
-		klog.Errorf("Failed to get all the pods (%v)", err)
-	} else {
-		ips = make([]net.IP, 0, len(existingPods))
-		for _, pod := range existingPods {
-			if pod.Status.PodIP != "" && !pod.Spec.HostNetwork {
-				podIPs, err := util.GetAllPodIPs(pod)
+	if config.Kubernetes.HostNetworkNamespace != "" &&
```

Comment:
again might be good to move this out into a subroutine.

It's also annoying that this logic exists in two different places...

Reply:
this will be called from exactly one place in it's current form. Yes, there will be two or three lines of re-use from the code when a single node gets added. But there isn't really much value in adding it in a separate sub-routine.